authentication/subscription/service/oso_subscription_service.go

```diff
@@ -170,10 +170,50 @@ func (s *osoSubscriptionServiceImpl) loadSubscriptions(ctx context.Context, user
 		log.Error(ctx, map[string]interface{}{
 			"err":         err,
 			"reg_app_url": regAppURL,
+			"username":    username,
 			"body":        bodyString,
 		}, "unable to unmarshal json with subscription status")
 		return nil, autherrors.NewInternalError(ctx, err)
 	}
 
 	return &sbs, nil
 }
+
+// DeactivateUser deactivates the user on OpenShift
+func (s *osoSubscriptionServiceImpl) DeactivateUser(ctx context.Context, username string) error {
+	// Load status from OSO
+	regAppURL := fmt.Sprintf("%s/api/accounts/%s/deprovision_osio?authorization_username=%s",
+		s.config.GetOSORegistrationAppURL(), username, s.config.GetOSORegistrationAppAdminUsername())
+	log.Debug(ctx, map[string]interface{}{
+		"url": regAppURL,
+	}, "calling remote registration application to check the user status")
+	req, err := http.NewRequest("GET", regAppURL, nil)
```

Comment:
`deprovision_osio` is **POST** and not **GET**

Reply:
Good catch. Can you also double check the expected response status?

Reply:
@alexeykazakov plz check conversation https://github.com/fabric8-services/fabric8-auth/pull/814#discussion_r273973364

Reply:
good catch indeed! Let me fix that

Reply:
done in d770717